src/main/java/com/servicewizard/resource/DocumentationResource.java

```diff
@@ -0,0 +1,38 @@
+package com.servicewizard.resource;
+
+import com.servicewizard.ServiceLocator;
+import com.servicewizard.generation.HtmlDocGenerator;
+import com.servicewizard.model.Service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+
+public abstract class DocumentationResource {
+
+    protected String getDocumentation() {
+        return documentation;
+    }
+
+    public DocumentationResource(ServiceLocator locator) {
+        List<Service> services = locator.locate();
+        cacheDocumentation(services);
+    }
+
+    private void cacheDocumentation(List<Service> services) {
```

Comment:
@forana I don't normally do such a thing, but it felt like I should move "complicated" "functionality" out of the "constructor".

"Constructor" shouldn't be quoted, it's an actual constructor.